kbs/src/attestation/backend.rs

```diff
@@ -268,31 +290,52 @@ impl AttestationService {
             }
 
             let attestation_str = serde_json::to_string_pretty(&attestation)
+                .inspect_err(|_| ATTESTATION_ERRORS.inc())
                 .context("Failed to serialize Attestation")?;
             debug!("Attestation: {attestation_str}");
 
             (session.request().tee, session.challenge().nonce.to_string())
         };
 
-        let attestation_str =
-            serde_json::to_string(&attestation).context("serialize attestation failed")?;
-        let token = self
-            .inner
-            .verify(tee, &nonce, &attestation_str)
-            .await
-            .context("verify TEE evidence failed")?;
-
         let mut session = self
             .session_map
             .sessions
             .get_async(session_id)
             .await
-            .ok_or(anyhow!("session not found"))?;
+            .ok_or(anyhow!("session not found"))
+            .inspect_err(|_| ATTESTATION_ERRORS.inc())?;
         let session = session.get_mut();
 
+        let tee_type_label = {
+            // it seems impossible to prevent serde from putting double-quotes
+            // around the tee name, get rid of them in a subsequent step
+            let quoted = serde_json::to_string(&session.request().tee)?;
+            quoted[1..quoted.len() - 1].to_owned()
+        };
+
+        let attestation_str = serde_json::to_string(&attestation)
+            .inspect_err(|_| ATTESTATION_ERRORS.inc())
+            .context("serialize attestation failed")?;
+        let token = self
+            .inner
+            .verify(tee, &nonce, &attestation_str)
+            .await
+            .inspect(|_| {
+                ATTESTATION_SUCCESSES
+                    .with_label_values(&[&tee_type_label])
+                    .inc()
+            })
```

Comment:
We can move this out of the nested `inspect` to keep the chain short.

Reply:
done